src/printer.js

```diff
@@ -3260,6 +3273,84 @@ function hasLeadingOwnLineComment(text, node) {
   return res;
 }
 
+function hasNakedLeftSide(node) {
+  return (
+    node.type === "BinaryExpression" ||
```

Comment:
I'm not convinced that it is more readable with a set. This is a special case where there are a lot of chained ones, but most of the time, you check a few conditions, then more conditions on the child... So it's easier to read when everything is inlined like this.

Also, for the Set, I'm not convinced that it is going to be faster. In this case, we're just comparing either pointers (static strings are usually unique-ified) or first few characters of a string which both are really fast. You don't need to do a function call and allocation at the beginning of the program.

I'm pretty sure we have lower hanging fruit in terms of performance than the way we do those comparisons. And as @rattrayalex mentioned, if we're going that way, we should do it across the entire codebase.